src/main/java/org/folio/rest/impl/ProtectionHelper.java

```diff
@@ -0,0 +1,109 @@
+package org.folio.rest.impl;
+
+import static org.folio.invoices.utils.ErrorCodes.INVOICE_UNITS_NOT_FOUND;
+import static org.folio.invoices.utils.ErrorCodes.USER_HAS_NO_PERMISSIONS;
+import static org.folio.invoices.utils.HelperUtils.convertIdsToCqlQuery;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.folio.HttpStatus;
+import org.folio.invoices.rest.exceptions.HttpException;
+import org.folio.invoices.utils.ProtectedOperationType;
+import org.folio.rest.acq.model.AcquisitionsUnit;
+import org.folio.rest.acq.model.AcquisitionsUnitCollection;
+import org.folio.rest.tools.client.interfaces.HttpClientInterface;
+
+import io.vertx.core.Context;
+
+public class ProtectionHelper extends AbstractHelper {
+
+  public static final String ACQUISITIONS_UNIT_ID = "acquisitionsUnitId";
+  private AcquisitionsUnitsHelper acquisitionsUnitsHelper;
+
+  public ProtectionHelper(HttpClientInterface httpClient, Map<String, String> okapiHeaders, Context ctx, String lang) {
+    super(httpClient, okapiHeaders, ctx, lang);
+    acquisitionsUnitsHelper = new AcquisitionsUnitsHelper(httpClient, okapiHeaders, ctx, lang);
+  }
+
+  /**
+   * This method determines status of operation restriction based on unit IDs from {@link Invoice}.
+   * 
+   * @param unitIds list of unit IDs.
+   *
+   * @throws HttpException if user hasn't permissions or units not found
+   */
+  public CompletableFuture<Void> isOperationRestricted(List<String> unitIds, ProtectedOperationType operation) {
+    return isOperationRestricted(unitIds, Collections.singleton(operation));
+  }
+
+  /**
+   * This method determines status of operation restriction based on unit IDs from {@link Invoice}.
+   * 
+   * @param unitIds list of unit IDs.
+   *
+   * @throws HttpException if user hasn't permissions or units not found
+   */
+  public CompletableFuture<Void> isOperationRestricted(List<String> unitIds, Set<ProtectedOperationType> operations) {
+    if (CollectionUtils.isNotEmpty(unitIds)) {
+      return getUnitsByIds(unitIds).thenCompose(units -> {
+        if (unitIds.size() == units.size()) {
+          if (applyMergingStrategy(units, operations)) {
+            return verifyUserIsMemberOfInvoiceUnits(unitIds);
+          }
+          return CompletableFuture.completedFuture(null);
+        } else {
+          throw new HttpException(HttpStatus.HTTP_VALIDATION_ERROR.toInt(), INVOICE_UNITS_NOT_FOUND);
+        }
+      });
+    } else {
+      return CompletableFuture.completedFuture(null);
+    }
+  }
+
+  /**
+   * Check whether the user is a member of at least one group from which the related invoice belongs.
+   *
+   * @return list of unit ids associated with user.
+   */
+  private CompletableFuture<Void> verifyUserIsMemberOfInvoiceUnits(List<String> unitIdsAssignedToInvoice) {
```

Comment:
The method name is a little confusing, can you please add acquistionsunits, so its evident